### PR TITLE
WIP for Weekly-review

### PR DESCRIPTION
**Pull request by Anasauce**
Here is an incremental PR for generating weekly review stats:

- created a view that directs to summary page after login
- added "total items" stat to render on summary page
- migration for audit table to track updates
- added hook in update function to create audit instance whenever there is an update
- created logic to extract values for audit instance

What still needs to be done:

- write logic to create audit instance with update values
- implement moment.js to pull all audits from correct time frame for summary
- render audits in summary view

### CODE REVIEW

File: models/item.js [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+  const Item = sequelize.define('Item',
+    {
+      title: DataTypes.STRING,
+      description: DataTypes.TEXT,
+      completed: DataTypes.BOOLEAN,
+      parent_id: DataTypes.INTEGER,
+      user_id: DataTypes.INTEGER
+    },
+
+
```
Comment (jrob8577) [inline models/item.js]: Remove excess whitespace.

File: migrations/20161101213307-create-audit.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    return queryInterface.createTable('Audits', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      table_name: {
```
Comment (jrob8577) [inline migrations/20161101213307-create-audit.js]: I thought we changed the structure of this table yesterday?

File: public/javascripts/items.js [hunk 3]
```diff
@@ -14,8 +14,10 @@ const params = data =>
 
 const checkJsonForSuccessField = json => {
   if( json.success ) {
+    console.log('checking for json success');
```
Comment (jrob8577) [inline public/javascripts/items.js]: Please revert changes to this file, unrelated to feature.

File: routes/accounts.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const validateEmail = require( '../src/mail/validate_email' )
 
 
 const AUTH_OPTIONS = {
-  successRedirect: '/items',
+  successRedirect: '/items/weekly',
```
Comment (jrob8577) [inline routes/accounts.js]: Please revert changes to this file

File: routes/items.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ const router = express.Router()
 const { allItemsQuery, filteredItemsQuery, respondWithItems } = require( './items/item_response' )
 const { buildTree } = require( './items/tree_creation' )
 const findAllItems = require('./items/find_all_items')
+const grabUserStats = require('../src/weekly_report')
```
Comment (jrob8577) [inline routes/items.js]: Please use a more meaningful name - it feels odd to me that a module that is called `weekly_report` exports a method called `grabUserStats`.  How about `weeklyUserReport` or some such?

File: src/Auditor.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+const returnAuditOptions = (updateType, item, data_type) => {
```
Comment (jrob8577) [inline src/Auditor.js]: Please use lower case file names.  There's a lot of repetition in this file, let's see if we can clean it up.

File: src/weekly_report.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+const today = moment()
+const todaysDate = moment()
+const minusSevenDays = todaysDate.subtract(7, 'days')
+console.log('-7: ', minusSevenDays.format("dddd, MMMM Do YYYY, h:mm:ss a"), '\n');
+console.log('todaysDate: ', today.format("dddd, MMMM Do YYYY, h:mm:ss a"));
+
+const lengthOfUse = (User, where) =>
+  User.findOne({where})
+    .then(user => {
+    return Math.abs(todaysDate.diff(user.createdAt, 'days'))
```
Comment (jrob8577) [inline src/weekly_report.js]: Indentation off; please use single line implicit return when possible.  With `moment`, you will not need `Math.abs` etc.

File: src/weekly_report.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+const bulletsChanged = (auditRecords) => {
+  const changedThisWeek = auditRecords.filter(record => record.updatedAt >= minusSevenDays)
+  return changedThisWeek.length
+    // ([all items w/ user_id && updatedAt(<= today’s date - 7) ].length)
+}
+
+const totalBullets = (auditRecords) => {
+  const totalBullets = auditRecords.filter(record => {
+      return record.element_name === 'row' && record.createdAt >= minusSevenDays
```
Comment (jrob8577) [inline src/weekly_report.js]: We should be able to construct a query to get this information, instead of doing this in the code.

**Comment by Anasauce** [comment]
PLEASE DISREGARD THIS PR...sorry!!!!!!

**Pull request closed without merging**